### PR TITLE
Fix #3022 - validate + polar.bargap

### LINKED ISSUE #3022: Polar bargap looks invalid to Plotly.validate
I made this plot:
```js
Plotly.newPlot(gd,[
  {type:'barpolar',r:r,theta:th,marker:{color:c,colorscale:'Jet'},subplot:'polar2'},
  {type:'scatterpolar',mode:'markers',theta:mth,r:mr,marker:{size:ms,color:mc,colorscale:[[0,'#600'],[1,'#f00']],sizeref:0.3}}
], {
  polar2: {domain:{x:[0,1],y:[0,1]},bargap:0.5,hole:0.83,radialaxis:{visible:false},angularaxis:{ticks:'',showline:false,nticks:40}},
  polar: {bgcolor:'#d3d3d3',hole:0.2,domain:{x:[0.1,0.9],y:[0.1,0.9]},
    radialaxis:{gridcolor:'#fff',showline:false,showticklabels:false,ticks:''},
    angularaxis:{gridcolor:'#fff',nticks:40,showline:false,ticks:'',showticklabels:false}}
})
```
But although it works correctly, `Plotly.validate complains`:
```js
Plotly.validate(gd.data, gd.layout)
LOG: In layout, key polar2.bargap is not part of the schema
```

@etpinard comments:
> ah right, that's a first
> a trace layout attribute inside a subplot container

(cc @Kully - you can ignore that one if you run `Plotly.validate`)

### PR DESCRIPTION
**Pull request by etpinard**
@alexcjohnson @antoinerg this solution isn't as strict as it could be (see comment in test), but at least it fixes the false-negative case of #3022.

### CODE REVIEW

File: test/jasmine/tests/validate_test.js [hunk 1]
```diff
@@ -539,4 +539,36 @@ describe('Plotly.validate', function() {
...
+            out[0], 'unused', 'layout', null, ['polar3'], 'polar3',
+            'In layout, container polar3 did not get coerced'
+        );
+
+        // Plotly.validate should be more strict here.
+        //
+        // It should log an 'unused' warning for `polar2.bargap`,
+        // but trace layout attribute set in subplot containers are considered
+        // valid as long as one trace on that graph has those trace layout
+        // attributes.
```
Comment (alexcjohnson) [inline test/jasmine/tests/validate_test.js]: In fact it's not just `Plotly.validate` that should be more strict - `supplyDefaults` coerces `polar2.bargap` unnecessarily, and if it didn't do that, `Plotly.validate` would at least return `'unused'` - which you might even argue is a better message than `'schema'` in this case, but that's a secondary concern. Might be worthwhile fixing this for `supplyDefaults` for the editor's sake.

Comment (etpinard) [reply test/jasmine/tests/validate_test.js]: Thanks for the tip.

Done in -> https://github.com/plotly/plotly.js/pull/3023/commits/c5cf89b71cd091a5b742fb2a80618672cce34cc7

File: src/plots/polar/legacy/axis_attributes.js [hunk 2]
```diff
@@ -13,6 +13,8 @@ var axesAttrs = require('../../cartesian/layout_attributes');
 var extendFlat = require('../../../lib/extend').extendFlat;
 var overrideAll = require('../../../plot_api/edit_types').overrideAll;
 
+var deprecationWarning = 'Legacy polar charts are deprecated!';
```
Comment (etpinard) [inline src/plots/polar/legacy/axis_attributes.js]: Should _fix_ https://github.com/plotly/documentation/issues/1095

File: src/plots/polar/legacy/area_attributes.js [hunk 3]
```diff
@@ -10,15 +10,48 @@
...
 var scatterAttrs = require('../../../traces/scatter/attributes');
 var scatterMarkerAttrs = scatterAttrs.marker;
+var extendFlat = require('../../../lib/extend').extendFlat;
+
+var deprecationWarning = 'Area traces are deprecated!';
 
 module.exports = {
-    r: scatterAttrs.r,
-    t: scatterAttrs.t,
+    r: extendFlat({}, scatterAttrs.r, {
```
Comment (etpinard) [inline src/plots/polar/legacy/area_attributes.js]: Follow-up on https://github.com/plotly/plotly.js/pull/3023/commits/1c0d3ed0512ef826bf2e6dc9d576bccfff2f18f3

Comment (alexcjohnson) [reply src/plots/polar/legacy/area_attributes.js]: Ah great, nice pattern for adding these warnings!

`scatterAttrs.r` already [asks users to switch to `scatterpolar`](https://github.com/plotly/plotly.js/blob/2a3dc2656d3c832ceb24b3c22c9155aa02fd7c7a/src/traces/scatter/attributes.js#L548) - so here we should switch that so the deprecation warning points people to `barpolar`. Similarly for the layout attrs, can we point people to `polar` subplots?

Comment (etpinard) [reply src/plots/polar/legacy/area_attributes.js]: new attempt: https://github.com/plotly/plotly.js/pull/3023/commits/019bacfc88069f51133f624c49063e5b01c94d58

**Comment by alexcjohnson** [comment]
Love it. :dancer:

**Pull request merged**